### PR TITLE
Make suite compatible with Windows

### PR DESCRIPTION
**Pull request by harigopal**

### CODE REVIEW

File: shared/todo.test.js [hunk 1]
```diff
@@ -13,56 +12,67 @@ beforeEach(() => {
...
-$ ./todo.sh done NUMBER      # Complete a todo
-$ ./todo.sh help             # Show usage
-$ ./todo.sh report           # Statistics
-`;
+$ ./todo.exe add "todo item"  # Add a new todo
+$ ./todo.exe ls               # Show remaining todos
+$ ./todo.exe del NUMBER       # Delete a todo
+$ ./todo.exe done NUMBER      # Complete a todo
+$ ./todo.exe help             # Show usage
+$ ./todo.exe report           # Statistics`;
```
Comment (jcsherin) [inline shared/todo.test.js:24]: @harigopal Is it possible to alias `todo.exe` to `todo` in windows. Similarly the `todo.sh` can also be shortened to just `todo`. Otherwise the `.exe` extension would be have to be explained to those who are submitting from *nixes. The usage test does not match the actual calling convention using `todo.sh`. Will an alias work?

File: shared/todo.test.js [hunk 2]
```diff
@@ -13,56 +12,67 @@ beforeEach(() => {
...
+  let received = execSync(
+    todoTxtCli("add", '"the thing i need to do"')
+  ).toString("utf8");
 
-  expect(received).toBe(expected);
+  expect(received).toEqual(expect.stringContaining(expected));
 });
 
 test("show error message when add is not followed by a todo", () => {
-  let expected = "Error: Missing todo string. Nothing added!" + EOL;
```
Comment (jcsherin) [inline shared/todo.test.js:47]: In `Node.js` the test will fail if you write:

```
console.log("Error: Missing todo string. Nothing added!");
```

We shouldn't remove the `EOL`?

File: shared/todo.bat [hunk 3]
```diff
@@ -0,0 +1 @@
+@echo off
```
Comment (jcsherin) [inline shared/todo.bat:1]: Hello, bat file 👍

File: shared/todo.test.js (hunk 2 above)
Comment (harigopal) [reply shared/todo.test.js:47]: I didn't get any test failures on either Ubuntu or Windows. 🤔 

Also, why would it fail? I've updated all tests to check for the _inclusion_ of the minimum required string, and not to look for an exact match.

I did this because console output on Windows & *nix were not exactly identical in all circumstances - mainly the addition or absence of _EOL_-s. So I got rid of manually added _EOL_-s from check-strings entirely.

File: shared/todo.test.js (hunk 1 above)
Comment (harigopal) [reply shared/todo.test.js:24]: The reason I used `exe` was to simply _indicate_ that this is how the _executable_ is supposed be used. Since there is no unified executable (`sh` on *nix, and `bat` on Windows), I thought it made sense to avoid favoring one platform over another.

I've tested symbolic links, and it appears to work. We can remove the extension from the test's expectation for the help command, and add an instruction in the README to create a symbolic link? Something along the lines of:

> ### Create symbolic link to the executable file
>
> Before running tests, create a symbolic link to the executable used on your platform - `todo.bat` on Windows, and `todo.sh` on *nix systems:
>
> **Windows:**
> ```
> > mklink todo todo.bat
> ```
>
> ***nix:**
> ```
> $ ln -s todo.sh todo
> ```

File: shared/todo.test.js (hunk 2 above)
Comment (jcsherin) [reply shared/todo.test.js:47]: Ok I missed the change of matcher to `expect.stringContaining`. This looks good then.

File: shared/todo.test.js (hunk 1 above)
Comment (jcsherin) [reply shared/todo.test.js:24]: Sounds good. Can you add this section to the README.

**Review by jcsherin — APPROVED** [review]
